packages/noir-ethereum-api/src/noir/oracles/server.ts

```diff
@@ -1,32 +1,41 @@
-import { JSONRPCServer, TypedJSONRPCServer } from 'json-rpc-2.0';
-import express from 'express';
-import bodyParser from 'body-parser';
+import { JSONRPCRequest, JSONRPCServer, TypedJSONRPCServer } from 'json-rpc-2.0';
+import Fastify from 'fastify';
 import { getHeaderHandler, getAccountHandler, JSONRPCServerMethods, ServerParams } from './server/handlers.js';
 import { createDefaultClient } from '../../ethereum/client.js';
 
-const PORT = 5555; // Suggested by nargo docs: https://noir-lang.org/docs/how_to/how-to-oracles#step-3---usage-with-nargo
+const PORT = 5555;
 const client = createDefaultClient();
 
 const server: TypedJSONRPCServer<JSONRPCServerMethods, ServerParams> = new JSONRPCServer();
 const serverParams = { client };
 server.addMethod('get_header', getHeaderHandler);
 server.addMethod('get_account', getAccountHandler);
 
-const app = express();
-app.use(bodyParser.json());
-app.post('/', (req, res) => {
-  const jsonRPCRequest = req.body;
-  console.log(jsonRPCRequest.method);
+const app = Fastify({
+  logger: {
+    transport: {
+      target: 'pino-pretty',
+      options: {
+        translateTime: 'SYS:standard',
+        ignore: 'pid,hostname,req.hostname,req.remoteAddress,req.remotePort'
+      }
+    }
+  }
+});
+
+app.post('/', (request, reply) => {
+  const jsonRPCRequest = request.body as JSONRPCRequest;
+  request.log.info({ jsonRPCRequest }, 'Received request');
 
   server.receive(jsonRPCRequest, serverParams).then((jsonRPCResponse) => {
     if (jsonRPCResponse) {
-      res.json(jsonRPCResponse);
+      reply.send(jsonRPCResponse);
     } else {
-      res.sendStatus(204);
+      reply.status(204).send();
     }
   });
 });
```

Comment:
We need to have A route. We can name it /oracles, but because this server only does that - I think it makes sense to not give it a name at all and use the root route

Reply:
How about it returns 404 then?